### ISSUE TITLE
node.js update

### ISSUE THREAD

**Issue by martindinicolino** [davide125/hassio-addons#7]
`The plugin "homebridge-switchbot" requires Node.js version of ^14.21.1 || ^16.18.1 || ^18.12.1 which does not satisfy the current Node.js version of v16.18.0. You may need to upgrade your installation of Node.js - see https://homebridge.io/w/JTKEF`

Updating node.js via the homebridge terminal (following the instructions here https://github.com/homebridge/homebridge/wiki/How-To-Update-Node.js) runs smoothly and I‘m prompted to restart homebridge. After a restart, node.js is updated to `v18.12.1` and I can install the updated version of the plugin.

The problem is that every time the container is restarted (Home Assistant is restarts/reboots), node.js is rolled back to `v16.18.0` and i receive the same prompt as before`The plugin "homebridge-switchbot" requires Node.js version of ^14.21.1 || ^16.18.1 || ^18.12.1 which does not satisfy the current Node.js version of v16.18.0. You may need to upgrade your installation of Node.js - see https://homebridge.io/w/JTKEF `

is there a way for me to definitively update the node.js version, or does that need to be done through an update to this addon?

**Comment by martindinicolino** [comment]
I was able to update node.js via the homebridge terminal following the instructions here https://github.com/homebridge/homebridge/wiki/How-To-Update-Node.js - this runs smoothly and updates node.js, but the active version of node.js remains unchanged according to the UI even after restarting homebridge

the terminal then suggested running `hb-service rebuild —all` (use after updating Node.js) - this generates the error `Cannot find module yallist` , but the active version of node.js was updated according to the UI

when I try to update the plugin mentioned above now I get the same error `Cannot find module yallist` 

so basically I messed up - any help would be highly appreciated

I can provide more info (logs) if needed

**Comment by davide125** [comment]
I've pushed a new version that updates the Homebridge container to the latest release, can you give this another try and let me know if it solved your issue? Thanks!

**Comment by davide125** [comment]
I've confirmed that the current version has node.js v18.13.0 so you should be good to go.

**Comment by martindinicolino** [comment]
Hello, thank you for the add-on and thank you for previously updating it to the latest node.js release.

Unfortunately as plugins are updated they keep requiering the latest version of node.js, which if i understand correctly can not be updated by the user of this add-on.

My homebridge plugins are now requiring 18.16.0 at least (the current version is 18.17.1).

When trying to update node.js, i keep running into the issue described above, where node.js is reset after every restart.

Would it be possible to implement a solution for a user to keep node.js updated, without requiering an update to this add-on?

**Comment by davide125** [comment]
I don't think there's an easy way to make node user-updateable and persist the update. I should have some time to get this addon updated again in a few weeks.

**Comment by martindinicolino** [comment]
Thank you for your reply and your time.
